### ISSUE TITLE
Enable secured Dashboard in Kubernetes through CRDs

### ISSUE THREAD

**Issue by dduportal** [containous/traefik#5626]
### Do you want to request a *feature* or report a *bug*?

Feature

### What did you expect to see?

[Must Have]

As a Kubernetes User
With Traefik v2.0 installed as "Ingress" with CRDs capability enabled
When I enable the dashboard in [secure mode](https://docs.traefik.io/v2.0/operations/dashboard/#secure-mode) (e.g. without the flag `--api.insecure`and without exposing the port `8080` of Traefik's pod) 
Then I want to define access to the Dashboard using CRDs (`IngressRoute` and eventually `Middleware` for security)

[Nice to have]

The [Traefik v2 Helm Chart](https://github.com/containous/traefik-helm-chart) should enable the dashboard with this CRD definition.

N.B. Unless I'm wrong, the issue #5242 is required to allow this behavior

### Workaround

With version `2.0.2`, you can only enable the dashboard in secure mode by enabling the file provider and use a ConfigMap to define TOML or YAML dynamic configuration for the dashboard/API router associated to the internal service `api@internal`.

**Comment by grozan** [comment]
was on the point to open an issue for the same thing.

The following Ingressroute does NOT work
```
apiVersion: traefik.containo.us/v1alpha1
kind: IngressRoute
metadata:
  name: dashboard
  namespace: traefik
spec:
  entryPoints:
    - web
  routes:
  - match: Host(`traefik.app-test.home`)
    kind: Rule
    services:
    - name: api@internal
```

The error message is 
```
time="2019-10-12T15:04:29Z" level=error msg="Cannot create service: service not found traefik/api@internal" providerName=kubernetescrd ingress=dashboard namespace=traefik serviceName=api@internal servicePort=0
```

There is no service called `api@internal` in the `traefik` namespace, of course. 
Looks to me as if the problem comes from the LoadServers function which tries to get the service definition from the namespace the CR is declared into.
https://github.com/containous/traefik/blob/master/pkg/provider/kubernetes/crd/kubernetes_http.go#L189
I guess `api@internal` should be treated differently

**Comment by nthypes** [comment]
We are having this same error!

**Comment by titansmc** [comment]
@dduportal So can we have file provider and kubernetes provider? how would the files look like?
Thanks in advance.

**Comment by nthypes** [comment]
@titansmc this is already possible today.

```
providers:
  kubernetesCRD: {}
  kubernetesIngress: {}
  file:
    filename: "/config/traefik/traefik-dynamic.yaml"
```

In this scenario you have Kubernetes Ingress + Kubernetes CRD + File providers.

Below an example of a cross-reference usage between providers:

An router defined in dynamic ocnfiguration file using Kubernetes CRD middleware.

```
http
  routers:
    https-redirect:
      rule: HostRegexp(`{host:.+}`)
      service: none
      entryPoints:
        - http
      middlewares:
        - traefik-https-redirect@kubernetescrd
```

**Comment by nthypes** [comment]
**PS:** Even that the cross-provider support works, the problem of this current issue persists.
 
We can't do `service: internal@api` on a Kubernetes CRD Route, because Traefik kinda of "concatenates" the Kubernetes namespace with the service name as @grozan pointed out.

**Comment by denysvitali** [comment]
I'm working on it. I'll propose a PR in the following hours

**Comment by ldez** [comment]
The problem is fixed in 2.1 by #5711

**Comment by denysvitali** [comment]
Oh, nevermind. Thanks 🎉
